config/settings.py

```diff
@@ -150,9 +152,23 @@ def pipe_delim(pipe_string):
 
 # Static files (CSS, JavaScript, Images)
 # https://docs.djangoproject.com/en/3.1/howto/static-files/
+# https://docs.djangoproject.com/en/3.1/ref/contrib/staticfiles/
 
 STATIC_URL = "/static/"
 
+# When running with DEBUG=False, the webserver needs to serve files from this location
+# python manage.py collectstatic has to be run to collect all static files into this location
+# The files need to served in brotli or gzip compressed format
+STATIC_ROOT = os.path.join(BASE_DIR, 'static/')
+
+# Location of client build
+CLIENT_BUILD_LOCATION = os.path.join(BASE_DIR, 'client_build/')
```

Comment:
Why not use `STATIC_ROOT` for this? Why have a separate setting?

Reply:
Django looks at directories in `STATICFILES_DIRS` to find static assets. It also does not allow `STATIC_ROOT` to be placed within `STATICFILES_DIRS`.

The client build location has to be present in `STATICFILES_DIRS` inorder for the assets to be served when DEBUG=False. So I've used a different location.

As an added note: When serving with DEBUG=False, Django does not serve assets on it's own, so we'll have to collate all static files into the STATIC_ROOT folder, and configure the webserver we intend to use, to serve those files.

Reply:
Sorry, I didn't phrase my question well. I meant, why not build the client into the existing `STATIC_ROOT` folder?

Reply:
Django cannot serve the STATIC_ROOT folder, so if we built the client into the STATIC_ROOT folder, it will not be possible to run it locally with Debug=True, without vite server.

It would work if we use Debug flag to determine the mode as Development or Production, but we are using a separate flag for it.

In short: We will not be able to run Mathesar as Debug=True and Mode=Production.

Reply:
This is the current behaviour:

Debug=False, Mode=Development (Vite server runs) [Can be run locally]
Debug=True, Mode=Develoment (Vite server runs) [Can be run locally]
Debug=False, Mode=Production (Client is built. But cannot be served, unless we collate files and configure webserver to serve the files) [**Cannot be run locally**]
Debug=True, Mode=Production (Client is built. Files are served by Django) [Can be run locally]

Reply:
Thanks for the explanation, makes sense.